### PR TITLE
add node-module option for node.__file/dirname

### PR DESCRIPTION
**Pull request by iamyamakin**
**What kind of change does this PR introduce?**

feature

closes https://github.com/webpack/webpack/issues/14072

**Did you add tests for your changes?**

yes

**Does this PR introduce a breaking change?**

no

**What needs to be documented once your changes are merged?**

new options node.__filename=node-module / node.__dirname=node-module

generate `__filename` and `__dirname` for common js modules when output.module=true to
`fileURLToPath(import.meta.url)` and `fileURLToPath(import.meta.url + "/..")` respectively

p.s. That is copy of https://github.com/webpack/webpack/pull/14247 with merge commit.

### CODE REVIEW

**Review by alexander-akait — APPROVED** [review]
Looks good for me, can you sign CLA?

**Comment by iamyamakin** [comment]
@alexander-akait done, something else?

**Comment by alexander-akait** [comment]
Thank you, release will be soon

File: lib/NodeStuffPlugin.js [hunk 1]
```diff
@@ -177,6 +218,13 @@ class NodeStuffPlugin {
 									"__dirname is a Node.js feature and isn't available in browsers."
 								);
 								break;
+							case "node-module":
+								setUrlModuleConstant(
+									"__dirname",
+									functionName =>
+										`${functionName}(import.meta.url + "/..").slice(0, -1)`
```
Comment (Hell1213) [inline lib/NodeStuffPlugin.js:225]: case "node-module":
    setUrlModuleConstant(
        "__dirname",
        functionName => {
            // Strip query parameters and fragments from import.meta.url
            const cleanUrl = functionName + "(import.meta.url.split('?')[0].split('#')[0])";
            // Append "/.." and remove trailing "/"
            return `${cleanUrl} + "/..").slice(0, -1)`;
        }
    );
    break;

**Pull request merged**